bavi/modules/url.py

```diff
@@ -0,0 +1,109 @@
+import logging
+import requests
+from lxml import html
+from bs4 import UnicodeDammit
+from urllib.parse import urlparse
+import re
+
+log = logging.getLogger('bavi.modules.url')
+
+URL_REGEX = r'(https?:\/\/(www\.)?[-a-zA-Z0-9@:%._\+~#=\u263a-\U0001f645]{1,256}\.[a-z]{2,6}\b([-a-zA-Z0-9@:%_\+.~#?&//=]*))'
+
+IP_REGEX = r"(https?:\/\/([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])(?<!172\.(16|17|18|19|20|21|22|23|24|25|26|27|28|29|30|31))(?<!127)(?<!10)(?<!0)\.([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])(?<!192\.168)(?<!172\.(16|17|18|19|20|21|22|23|24|25|26|27|28|29|30|31))\.([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])\.([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])(?<!\.255$)([-a-zA-Z0-9@:%_\+.~#?&//=]*))"
```

Comment:
Is this `IP_REGEX` behemoth needed?  It seems like the `URL_REGEX` could be relaxed a bit to accomodate IPs.

I also note that your URL regex won't match `.technology` as a TLD, since I see `[a-z]{2,6}`.